pipelines/sqre/infra/build_sciplatlab.groovy

```diff
@@ -16,91 +16,61 @@ node('jenkins-manager') {
-    'IMAGE_NAME',
-    'NO_PUSH',
-    'FLATTEN',
-    'VERBOSE',
-    'JLBLEED',
     'TAG',
-    'TAG_PREFIX',
+    'SUPPLEMENTARY',
+    'NO_PUSH',
+    'BRANCH',
     'TIMEOUT',
   ])
 
   String tag         = params.TAG
-  Boolean jlbleed    = params.JLBLEED
-  Boolean pushDocker = (! params.NO_PUSH.toBoolean())
-  Boolean flatten    = params.FLATTEN.toBoolean()
-  Boolean verbose    = params.VERBOSE.toBoolean()
-  String baseImage   = params.BASE_IMAGE
-  String imageName   = params.IMAGE_NAME
-  String tagPrefix   = params.TAG_PREFIX
+  String supplementary = params.SUPPLEMENTARY
   Integer timelimit  = params.TIMEOUT
+  String branch = params.BRANCH
+  Boolean pushDocker = (! params.NO_PUSH.toBoolean())
+
+  def make = { String m_target, String m_args ->
+    util.bash """
+    make '${m_target}' '${m_args}'
+    """
+  } // make
 
-  def run = {
+  def run = { String branch ->
     stage('checkout') {
-      def branch = 'prod'
-      if (jlbleed) {
-        branch = 'master'
-      }
       git([
-        url: 'https://github.com/lsst-sqre/nublado',
+        url: 'https://github.com/lsst-sqre/sciplat-lab',
         branch: branch
       ])
     }
 
     // ensure the current image is used
     stage('docker pull') {
-      docImage = "${baseImage}:${tagPrefix}${tag}"
+      docImage = "lsstsqre/centos:7-stack-lsst_distrib-${tag}"
       docker.image(docImage).pull()
     }
 
-    stage('build+push') {
-      def opts = ''
-      if (jlbleed) {
-        opts = '-e -s jlbleed ${opts}'
-      }
-      if (flatten) {
-          opts = "-f ${opts}"
-      }
-      if (verbose) {
-          opts = "-v ${opts}"
+    stage('execute') {
+      def m_args = 'tag=${tag}'
+
+      if (supplementary) {
+        m_args = '${m_args} supplementary=${supplementary}'
       }
-      opts=opts.trim()
-      dir('jupyterlab') {
-        if (pushDocker) {
-          docker.withRegistry(
-            'https://index.docker.io/v1/',
-            'dockerhub-sqreadmin'
-          ) {
-            util.bash """
-              ./bld \
-               -b '${baseImage}' \
-               -n '${imageName}' \
-               -t '${tagPrefix}' \
-               ${opts} \
-               '${tag}'
-            """
+      if (pushDocker) {
+        docker.withRegistry(
+          'https://index.docker.io/v1/',
+          'dockerhub-sqreadmin'
+        ) { // build and push
+            make('push', m_args)
           }
-        } else {
-          util.bash """
-              ./bld \
-               -x \
-               -b '${baseImage}' \
-               -n '${imageName}' \
-               -t '${tagPrefix}' \
-               ${opts} \
-               '${tag}'
-              docker build .
-          """
-        }
-      }
-    }
+      } else { // just build
+        make('image', m_args)
+      } // pushDocker clause
+    } // execute
   } // run
 
   util.nodeWrap('docker') {
     timeout(time: timelimit, unit: 'HOURS') {
-      run()
+      run(branch)
```

Comment:
Should non-prod branch builds push the same container tags as prod ones?  If they're for testing, perhaps forcing a SUPPLEMENTARY string for them might be appropriate?

Reply:
Ooooh, good idea.  They should not.